backend/models/grades.model.js

```diff
@@ -0,0 +1,60 @@
+const mongoose = require('mongoose');
+
+const gradeSchema = new mongoose.Schema(
+    {
+        sem1: {
```

Comment:
add `userId` as a reference to the User model so that we can map the user with the grades